### PR TITLE
Add url, progress and duration for iOS parity

### LINKED ISSUE #35: Android get URL of current track like iOS
Hey. 

I'm instantiating a list of player on a react native page.
The goal is to display a list of question and let the user play the response of the question.

In peculiar if the user play a question, then click another question, it should stop the first and start the second.

When the user click on the question, onPress is called.

```js
onAudioEvt = (evt: any) => {
    switch (evt.status) {
      case PLAYING:
        if (evt.url === this.getUrl()) {
          this.setState({ playing : true });
        } else {
          this.setState({ playing : false });
          this.subscription.remove();
          delete this.subscription;
        }
        break;
      case STOPPED:
      case ERROR:
        this.setState({ playing : false });
        this.subscription.remove();
        delete this.subscription;
        break;
    }
  }

  onPress = () => {
    if (!this.subscription) {
      this.subscription = DeviceEventEmitter.addListener('AudioBridgeEvent', this.onAudioEvt);
      ReactNativeAudioStreaming.play(this.getUrl(), {});
    } else {
      ReactNativeAudioStreaming.stop();
    }
  }
```

In ios this works fine because evt contain an url props (and also a duration that I ulse elsewhere but that is an other issue.

Is there a way on android to get the event current url ??

### PR DESCRIPTION
**Pull request by tychota**
Fix #35

### CODE REVIEW

**Comment by tychota** [comment]
I will test using my project that all works with this commit hash then let you merge.
You can start review it.

File: android/src/main/java/com/audioStreaming/Signal.java [hunk 1]
```diff
@@ -300,24 +312,48 @@ public void playerPCMFeedBuffer(boolean isPlaying, int bufSizeMs, int bufCapacit
...
 
     @Override
     public void playerException(final Throwable t) {
         this.isPlaying = false;
         this.isPreparingStarted = false;
+        StringWriter sw = new StringWriter();
+        PrintWriter pw = new PrintWriter(sw);
+        t.printStackTrace(pw);
+        Intent BroadcastIntent = new Intent(Mode.ERROR);
+        BroadcastIntent.putExtra("stacktrace", sw.toString());
```
Comment (tlenclos) [inline android/src/main/java/com/audioStreaming/Signal.java]: BroadcastIntent is not used here and I don't think we should pass this much data over the bridge. We should create custom error code maybe, but this is an other issue. Could you remove the stacktrace part ?

Comment (tychota) [inline android/src/main/java/com/audioStreaming/Signal.java]: Forgot to remove dead code. 

Nowadays, when there is an error that is hard to see why (some of my file were private on AWS and I took me some time). But I agree, it is another feature.

I'm gonna update this PR.

**Comment by tychota** [comment]
It is working perfectly in my project. I did not use getStatus thought, so if you can test that.

Thank for the help.

**Comment by brettpappas** [comment]
This will be a big help for getting Android closer to feature parity with iOS (ability to track progress). 

Looks like the 1 remaining thing is the ability to play the audio at a specific starting point. If we could do that with aacdecoder then we would be able to add the seekToTime, goForward, and goBack methods like we have for iOS. I use these methods to give the player a button to advance 30 seconds, rewind 30 seconds like below:

screen shot 2016-11-10 at 9 58 15 am

**Comment by tychota** [comment]
Any ETA before merge of this PR ?

I've sadly worked on the master branch of the fork and I think I will need it the to push some unrelated commits on the fork to be able to debug an issue with some stream on android.

If you need more time to review, I can rewrite the history to move this PR's commits into another branch and reopen a new PR with those commits.

**Comment by tlenclos** [comment]
@tychota you can just use an other branch based on master for your work if you want. I will not merge this before thursday sadly

**Comment by tychota** [comment]
Oki ! I will do that ;)

Thank for that pragamatic advice : I had not thought this solution :)

**Comment by tlenclos** [comment]
Hello there, I'm sorry but it does not seem to work.

I'm always getting a duration of 2.5 and a progress between 0.5 and 1.5.

**Comment by tychota** [comment]
Mmh. The progress stuff look weird on my side too.

I'm getting progress in percent (progress/duration) between 0 and 100% but they seems to oscillate.

I guess it does not take the full file duration as duration but rather the already streamed duration. One coworker also did also catch this today. Since we are playing 1 minute file but over great network, I guess I didn't catch this earlier.

Now I'm lost. I don't know what are exactly the information I'm getting but I guess it is only what's buffered. It is clearly not the same as in iOS. And i don't know how to obtain feature parity.

Is it ever possible with aacPlayer ?

**Comment by tychota** [comment]
Shall I remove the progress part, open another PR with only url and try to get something smaller merged ?

Or do you think this will be lost of time ?

**Comment by tlenclos** [comment]
Don't waste your time on this. The url is something that comes from user side so this is not really a problem. I will add it later after the migration.

**Comment by tychota** [comment]
So let' close :)

Sorry that it did not end up being merged :/

**Pull request closed without merging**